src/common/config.cc

```diff
@@ -949,6 +953,10 @@ int md_config_t::_get_val(const char *key, char **buf, int len) const
     ostringstream oss;
     if (bool *flagp = boost::get<bool>(&cval)) {
       oss << (*flagp ? "true" : "false");
+    } else if (float *fp = boost::get<float>(&cval)) {
```

Comment:
something like that, we can have two commits here. one for the refactor, the other for fixing the output format.

also, you don't need to use `strlen()` for the length of a `std::string`. you could just use `val.length()` instead. and please use `strncpy()` if possible.